src/SFML/Graphics/ImageLoader.cpp

```diff
@@ -227,35 +227,34 @@ bool ImageLoader::saveImageToFile(const std::string& filename, const std::vector
     if (!pixels.empty() && (size.x > 0) && (size.y > 0))
     {
         // Deduce the image type from its extension
-        if (filename.size() > 3)
+
+        // Extract the extension
+        const std::size_t dot = filename.find_last_of('.');
```

Comment:
Never really used `rfind()` (didn't actually know it :D). I guess for single characters the implementation is exactly the same. The only difference occurs when passing an actual string.